src/storage.rs

```diff
@@ -173,62 +173,64 @@ impl StorageBackend for CQLSession {
-            "nonce" => tx.nonce().to_string()
+            tx_hash.to_string(),
+            tx.payload().to_string(),
+            tx.address().to_string(),
+            tx.value().0 as i32,
+            tx.obsolete_tag().to_string(),
+            tx.timestamp().0 as i32,
+            tx.index().0 as i16,
+            tx.last_index().0 as i16,
+            tx.bundle().to_string(),
+            tx.trunk().to_string(),
+            tx.branch().to_string(),
+            tx.tag().to_string(),
+            tx.attachment_ts().0 as i32,
+            tx.attachment_lbts().0 as i32,
+            tx.attachment_ubts().0 as i32,
+            tx.nonce().to_string()
         );
         let bundle = query_values!(
-            "hash" => tx.bundle().to_string(),
-            "kind" => 0i8,
-            "timestamp" => tx.timestamp().0 as i32,
-            "tx" => tx_hash.to_string()
+            tx.bundle().to_string(),
+            0i8,
+            tx.timestamp().0 as i32,
+            tx_hash.to_string()
         );
         let address = query_values!(
-            "hash" => tx.address().to_string(),
-            "kind" => 1i8,
-            "timestamp" => tx.timestamp().0 as i32,
-            "tx" => tx_hash.to_string()
+            tx.address().to_string(),
+            1i8,
+            tx.timestamp().0 as i32,
+            tx_hash.to_string()
         );
         let tag = query_values!(
-            "hash" => tx.tag().to_string(),
-            "kind" => 2i8,
-            "timestamp" => tx.timestamp().0 as i32,
-            "tx" => tx_hash.to_string()
+            tx.tag().to_string(),
+            2i8,
+            tx.timestamp().0 as i32,
+            tx_hash.to_string()
         );
         let approvee1 = query_values!(
-            "hash" => tx_hash.to_string(),
-            "kind" => 3i8,
-            "timestamp" => tx.timestamp().0 as i32,
-            "tx" => tx.trunk().to_string()
+            tx_hash.to_string(),
+            3i8,
+            tx.timestamp().0 as i32,
+            tx.trunk().to_string()
         );
         let approvee2 = query_values!(
-            "hash" => tx_hash.to_string(),
-            "kind" => 3i8,
-            "timestamp" => tx.timestamp().0 as i32,
-            "tx" => tx.branch().to_string()
+            tx_hash.to_string(),
+            3i8,
+            tx.timestamp().0 as i32,
+            tx.branch().to_string()
         );
 
-        // TODO: Batch query instead, but it seems curretn btach executor is not compatible to scylla.
-        self.0.query_with_values(INSERT_TX_QUERY, values)?;
-        self.0.query_with_values(INSERT_EDGE_QUERY, bundle)?;
-        self.0.query_with_values(INSERT_EDGE_QUERY, address)?;
-        self.0.query_with_values(INSERT_EDGE_QUERY, tag)?;
-        self.0.query_with_values(INSERT_EDGE_QUERY, approvee1)?;
-        self.0.query_with_values(INSERT_EDGE_QUERY, approvee2)?;
-
+        let batch = BatchQueryBuilder::new()
+            .add_query(INSERT_TX_QUERY, values)
```

Comment:
So could you explain why add_query_prepare not working here?

Reply:
The add_query_prepare actually works. The previous errors happen due to the macro query_value!(). In the batch query, we should not provide the queries w/ value names. See descriptions in https://github.com/apache/cassandra/blob/trunk/doc/native_protocol_v4.spec#L416-L422